src/hooks/useRCAuth4Google.js

```diff
@@ -0,0 +1,74 @@
+import { useState, useContext } from 'react';
+import { useToastBarDispatch } from '@rocket.chat/fuselage-toastbar';
+import RCContext from '../context/RCInstance';
+import { useGoogleLogin } from './useGoogleLogin';
+import { useToastStore, useUserStore, useTotpStore } from '../store';
+
+export const useRCAuth4Google = () => {
+  const [userOrEmail, setUserOrEmail] = useState(null);
+
+  const { signIn, signOut } = useGoogleLogin(
```

Comment:
hooks should handle 'single responsibility' only, we should not have any other hook as its deps. This will help us reuse code in other projects as well

Reply:
sorry for that  i changed this now we only have a SignIn import